src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -297,9 +297,15 @@ private void LogCallEnd()
         }
         if (_status.StatusCode != StatusCode.OK)
         {
-            GrpcEventSource.Log.CallFailed(_status.StatusCode);
+            if (GrpcEventSource.Log.IsEnabled())
+            {
+                GrpcEventSource.Log.CallFailed(_status.StatusCode);
+            }
+        }
+        if (GrpcEventSource.Log.IsEnabled())
```

Comment:
since you're always going to test enabled here, is it worth pulling this into a local around L297, to avoid multiple checks?

Reply:
IsEnabled implementation directly returns a field. IsEnabled is so cheap (it's probably inlined) that I don't think it's worth it.